### PR TITLE
Increase pathlib.Path use

### PR DESCRIPTION
**Pull request by ahartikainen**
Switch to use pathlib.Path as the main way to handle path information.

### CODE REVIEW

**Comment by riddell-stan** [comment]
Is this ready to be reviewed?

**Comment by ahartikainen** [comment]
Yes (I forgot the PR 👍 )

File: httpstan/cache.py [hunk 1]
```diff
@@ -94,12 +96,10 @@ def dump_fit(fit_bytes: bytes, name: str) -> None:
         name: Stan fit name
         fit_bytes: LZ4-compressed messages associated with Stan fit.
     """
-    cache_path = appdirs.user_cache_dir("httpstan", version=httpstan.__version__)
     # fits are stored under their "parent" models
-    fits_path = os.path.join(*([cache_path] + name.split("/")[:-1]))
-    fit_filename = os.path.join(fits_path, f'{name.split("/")[-1]}.jsonlines.lz4')
-    os.makedirs(fits_path, exist_ok=True)
-    with open(fit_filename, mode="wb") as fh:
+    fit_filename = cache_directory() / (name + ".jsonlines.lz4")
```
Comment (riddell-stan) [inline httpstan/cache.py:100]: The previous version was a bit more explicit ("Explicit is better than implicit."). Also, can this use model_directory instead? In theory, the function should not need to know the cache directory. (The original was unskillful.)

File: httpstan/cache.py [hunk 2]
```diff
@@ -113,12 +113,10 @@ def load_fit(name: str) -> bytes:
     Returns
         LZ4-compressed messages associated with Stan fit.
     """
-    cache_path = appdirs.user_cache_dir("httpstan", version=httpstan.__version__)
     # fits are stored under their "parent" models
-    fits_path = os.path.join(*([cache_path] + name.split("/")[:-1]))
-    fit_filename = os.path.join(fits_path, f'{name.split("/")[-1]}.jsonlines.lz4')
+    fit_filename = cache_directory() / (name + ".jsonlines.lz4")
```
Comment (riddell-stan) [inline httpstan/cache.py:117]: Maybe we need a fits_directory function to clean this all up?

File: httpstan/cache.py [hunk 3]
```diff
@@ -130,7 +128,5 @@ def delete_fit(name: str) -> None:
     Arguments:
         name: Stan fit name
     """
-    cache_path = appdirs.user_cache_dir("httpstan", version=httpstan.__version__)
-    fits_path = os.path.join(*([cache_path] + name.split("/")[:-1]))
-    fit_id = name.split("/")[-1]
-    pathlib.Path(os.path.join(fits_path, f"{fit_id}.jsonlines.lz4")).unlink()
+    fit_filename = cache_directory() / (name + ".jsonlines.lz4")
```
Comment (riddell-stan) [inline httpstan/cache.py:131]: Another use for fits_directory

**Review by riddell-stan — COMMENTED** [review]
This looks good. Thanks for doing this.

How about adding a `fit_directory` function (which resembles `model_directory`). I think it will make the code more readable.

**Comment by riddell-stan** [comment]
@ahartikainen Any thoughts about my comments?

**Comment by ahartikainen** [comment]
Yes, I will fix the code when I have time. (I think this can go in in later release, no hurry right?)

File: httpstan/cache.py [hunk 4]
```diff
@@ -16,11 +15,23 @@
...
+
+def model_directory(model_name: str) -> Path:
     """Get the path to a model's directory. Directory may not exist."""
-    cache_path = appdirs.user_cache_dir("httpstan", version=httpstan.__version__)
     model_id = model_name.split("/")[1]
-    return os.path.join(cache_path, "models", model_id)
+    return cache_directory() / "models" / model_id
+
+
+def fit_filepath(fit_name: str) -> Path:
```
Comment (riddell-stan) [inline httpstan/cache.py:29]: I'd stick with `fit_path` ("filepath" almost never occurs in the Python documentation; I think people have settled on "path" as the term)

File: httpstan/cache.py [hunk 5]
```diff
@@ -94,12 +104,10 @@ def dump_fit(fit_bytes: bytes, name: str) -> None:
         name: Stan fit name
         fit_bytes: LZ4-compressed messages associated with Stan fit.
     """
-    cache_path = appdirs.user_cache_dir("httpstan", version=httpstan.__version__)
     # fits are stored under their "parent" models
-    fits_path = os.path.join(*([cache_path] + name.split("/")[:-1]))
-    fit_filename = os.path.join(fits_path, f'{name.split("/")[-1]}.jsonlines.lz4')
-    os.makedirs(fits_path, exist_ok=True)
-    with open(fit_filename, mode="wb") as fh:
+    fit_filename = fit_filepath(name)
```
Comment (riddell-stan) [inline httpstan/cache.py:108]: This should be `fit_path` not `fit_filename`, right?

File: httpstan/cache.py (hunk 4 above)
Comment (ahartikainen) [reply httpstan/cache.py:29]: What do you think about the function? I tried to make it a bit more explicit, but I was not sure which one is cleaner solution

Split fit to model_id, fit_id and then call model_directory or just get cache path.

File: httpstan/cache.py (hunk 5 above)
Comment (ahartikainen) [reply httpstan/cache.py:108]: Yes, fit_path sounds better

File: httpstan/cache.py (hunk 4 above)
Comment (riddell-stan) [reply httpstan/cache.py:29]: I don't have any objection to it. It's a tiny bit odd that we have model_directory but not fit_directory. On the other hand, if we never refer to fit_directory directly, there's no need for a function returning it.

**Review by riddell-stan — APPROVED** [review]
This looks great. Thank you!

This just needs to be rebased now.

**Pull request merged**